### PR TITLE
Simplified

### PR DESCRIPTION
**Pull request by iNoles**
#25 I just started to do this. Is there is any more functions to be simplified for this?

### CODE REVIEW

**Comment by WeaponMan** [comment]
`XmlReader.skipTo(String toFind)` theoretically can be inlined as 
```java
long endPosition = source.indexOf('>');
if(endPosition == -1){
 throw syntaxError("Error");
}
source.skip(endPosition);
```

**Comment by WeaponMan** [comment]
Why did you closed this?

**Comment by WeaponMan** [comment]
Just change `offset` in `rangeEquals` to `0` and it will pass the tests.
If you can continue with this PR.

**Comment by iNoles** [comment]
@WeaponMan Well, I am working on the local branch first into a new branch.

**Comment by WeaponMan** [comment]
@iNoles ok, just squash commits into single one it will look better ;)

**Comment by iNoles** [comment]
@WeaponMan Thanks for your Suggestions.

**Comment by WeaponMan** [comment]
I have managed to get rid off skipTo method:
```java
if (peekStack == XmlScope.NONEMPTY_DOCUMENT && isDocTypeDefinition()) {
  long endingIndex = source.indexOf((byte) '>', DOCTYPE_OPEN.size() - 1);
  if (endingIndex == -1) {
    throw syntaxError("Unterminated <!DOCTYPE> . Inline DOCTYPE is not support at the moment.");
  }
  source.skip(endingIndex + 1);
  // TODO inline DOCTYPE.
  p = 0;
  continue;
} else if (peek == '!' && fillBuffer(4)) {
  long endingIndex = source.indexOf(COMMENT_CLOSE, 3);
  if (endingIndex == -1) {
    throw syntaxError("Unterminated comment");
  }
  source.skip(endingIndex + COMMENT_CLOSE.size());
  p = 0;
  continue;
} else if (peek == '?') {
  long endingIndex = source.indexOf(XML_DECLARATION_CLOSE, 1);
  if (endingIndex == -1) {
    throw syntaxError("Unterminated xml declaration or processing instruction \"<?\"");
  }
  source.skip(endingIndex + XML_DECLARATION_CLOSE.size());
  p = 0;
  continue;
}
```
Can you included this bit? If not a I will make another PR.

**Comment by iNoles** [comment]
@WeaponMan Sure.

**Comment by sockeqwe** [comment]
Thanks! 
Btw. manually squashing is usually not needed because github also does it automatically on merging pull request

File: core/src/main/java/com/tickaroo/tikxml/XmlReader.java [hunk 1]
```diff
@@ -300,42 +302,22 @@ private int doPeek() throws IOException {
...
-        && buffer.getByte(0) == '<'
-        && buffer.getByte(1) == '!'
-        && buffer.getByte(2) == 'D'
-        && buffer.getByte(3) == 'O'
-        && buffer.getByte(4) == 'C'
-        && buffer.getByte(5) == 'T'
-        && buffer.getByte(6) == 'Y'
-        && buffer.getByte(7) == 'P'
-        && buffer.getByte(8) == 'E';
+  	return buffer.rangeEquals(0, DOCTYPE_OPEN);
```
Comment (sockeqwe) [inline core/src/main/java/com/tickaroo/tikxml/XmlReader.java]: Nice! 👍

**Pull request merged**